src/overlays/actors/ovl_Obj_Ice_Poly/z_obj_ice_poly.c

```diff
@@ -32,37 +34,327 @@ const ActorInit Obj_Ice_Poly_InitVars = {
+                    actor = actor->next;
+                }
+            } while (actor != NULL);
+        }
+    } else if ((this->unk_149 != OBJICEPOLY_FF_FF) && Flags_GetSwitch(globalCtx, this->unk_149)) {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        this->unk_14A = 1;
+        this->actionFunc = func_80931E58;
+    } else {
+        if ((this->actor.parent != NULL) && (this->actor.parent->update != NULL)) {
+            this->actor.parent->freezeTimer = 40;
+        } else {
+            this->actor.parent = NULL;
+        }
+
+        if (this->unk_14A == 0) {
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliders1[0].base);
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliders1[1].base);
+        }
+
+        for (i = 0; i < ARRAY_COUNT(this->colliders1); i++) {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliders1[i].base);
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliders1[i].base);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliders2[i].base);
+        }
+    }
+
+    if (Rand_ZeroOne() < 0.05f) {
+        sp58 = Rand_ZeroOne();
+        if (Rand_ZeroOne() < 0.5f) {
+            phi_v0 = -1;
+        } else {
+            phi_v0 = 1;
+        }
+        sp6C.x = (phi_v0 * (15.0f + (sp58 * 15.0f)) * this->actor.scale.x) + this->actor.world.pos.x;
+        sp6C.y = (((Rand_ZeroOne() * 90.0f) + 10.0f) * this->actor.scale.y) + this->actor.world.pos.y;
+
+        sp58 = Rand_ZeroOne();
+        if (Rand_ZeroOne() < 0.5f) {
+            phi_v0 = -1;
+        } else {
+            phi_v0 = 1;
+        }
+        sp6C.z = (phi_v0 * (15.0f + (sp58 * 15.0f)) * this->actor.scale.z) + this->actor.world.pos.z;
+
+        EffectSsKiraKira_SpawnDispersed(globalCtx, &sp6C, &gZeroVec3f, &gZeroVec3f, &D_80932378, &D_8093237C, 2000, 5);
+    }
+}
+
+void func_80931E58(ObjIcePoly* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        if (this->unk_14A == 1) {
+            func_80931828(this, globalCtx);
+            Actor_MarkForDeath(&this->actor);
+        } else {
+            this->unk_14A = 40;
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_ICE_MELT);
+            this->actionFunc = func_80931EEC;
+        }
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
+
+void func_80931EEC(ObjIcePoly* this, GlobalContext* globalCtx) {
+    Vec3f spAC;
+    Vec3f spA0;
+    Vec3f sp94;
+    f32 temp_f20;
+    s32 i;
+    s32 phi_v0;
+
+    spAC.x = 0.0f;
+    spAC.y = this->actor.scale.y;
+    spAC.z = 0.0f;
+
+    spA0.x = 0.0f;
+    spA0.y = this->actor.scale.y;
+    spA0.z = 0.0f;
+
+    for (i = 0; i < 2; i++) {
+        temp_f20 = Rand_ZeroOne();
+        if (Rand_ZeroOne() < 0.5f) {
```

Comment:
wait... it calls ZeroOne twice and ignores the first one? (overwrites this later with _another one_?)

Reply:
It's not ignored, it's used in the expression below.

Reply:
oh I visually skipped over it, focusing on the re-use nvm